crypto/src/main/java/net/corda/v5/crypto/DigestAlgorithmName.java

```diff
@@ -0,0 +1,79 @@
+package net.corda.v5.crypto;
+
+import net.corda.v5.base.annotations.CordaSerializable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The digest algorithm name. This class is to be used in Corda hashing API.
+ */
+@CordaSerializable
+public final class DigestAlgorithmName {
+    @NotNull
+    private final String name;
+
+    /**
+     * Construct a digest algorithm name.
+     * <p>
+     * * @param name The name of the digest algorithm to be used for the instance.
+     */
+    public DigestAlgorithmName(@NotNull String name) {
+        if (name.isBlank()) throw new IllegalArgumentException("Hash algorithm name unavailable or not specified");
```

Comment:
You should probably check for name being `null` too, since that what Kotlin would have done.

Reply:
done